app/upgrade_cudos.go

```diff
@@ -1493,57 +1507,66 @@ func registerBalanceMovement(fromAddress, toAddress string, sourceAmount sdk.Coi
 	manifest.MoveMintedBalance.Movements = append(manifest.MoveMintedBalance.Movements, movement)
 }
 
-func registerManifestMoveDelegations(fromAddress, toAddress string, memo string, manifest *UpgradeManifest) {
+func registerManifestMoveDelegations(fromAddress, toAddress, validatorAddress string, amount sdk.Int, memo string, manifest *UpgradeManifest) {
 	if manifest.MoveDelegations == nil {
 		manifest.MoveDelegations = &UpgradeMoveDelegations{}
 	}
 
 	movement := UpgradeDelegationMovements{
-		From: fromAddress,
-		To:   toAddress,
-		Memo: memo,
+		From:      fromAddress,
+		To:        toAddress,
+		Validator: validatorAddress,
+		Tokens:    amount,
+		Memo:      memo,
 	}
 	manifest.MoveDelegations.Movements = append(manifest.MoveDelegations.Movements, movement)
 	manifest.MoveDelegations.NumberOfMovements = len(manifest.MoveDelegations.Movements)
 }
 
-func moveGenesisDelegations(genesisData *GenesisData, fromAddress, toAddress string, manifest *UpgradeManifest) error {
-	sourceDelegations, exists := genesisData.delegations.Get(fromAddress)
-
-	if !exists {
-		registerManifestMoveDelegations(fromAddress, toAddress, "no_delegations", manifest)
-		// Nothing to move
+func moveGenesisDelegation(genesisData *GenesisData, fromDelegatorAddress, toDelegatorAddress string, validatorAddress string, amount sdk.Int, manifest *UpgradeManifest, memo string) error {
+	// Nothing to move
+	if fromDelegatorAddress == toDelegatorAddress {
 		return nil
 	}
 
-	if destDelegation, destDelegationExists := genesisData.delegations.Get(toAddress); destDelegationExists {
-		// Add delegations - destination delegator has some delegations
-
-		// List all validator addresses in source delegation
-		for _, validatorAddr := range sourceDelegations.Keys() {
-			srcCoins := sourceDelegations.MustGet(validatorAddr)
+	sourceDelegations, exists := genesisData.delegations.Get(fromDelegatorAddress)
+	if !exists {
+		return fmt.Errorf("genesis source delegations of %s not found", fromDelegatorAddress)
```

Comment:
It kind of feels like this should not be an error case, but simply valid case for doing nothing = simple success return from the function.
```suggestion
		return nil
```

Reply:
Ignore the comment above, it is the philosophical bent - I checked as how the function is used